src/client/src/ui/blotter/blotterOperations.ts

```diff
@@ -19,27 +20,33 @@ export const blotterServiceEpic = (blotterService$, openFin) => (action$, store)
   return action$.ofType(REF_ACTION_TYPES.REFERENCE_SERVICE)
     .do(subscribeOpenFinToBlotterData(openFin, store))
     .flatMapTo(blotterService$.getTradesStream())
-    .map(fetchBlotter)
+    .map(createNewTradesAction)
 }
 
 export const blotterRegionsSettings = regionsSettings('Blotter', 850, 250, false)
 
-export const blotterServiceReducer = (state: any = {}, action) => {
+interface Trades {
+  [tradeId: number]: Trade
+}
+
+interface State {
+  trades: Trades
+}
+
+const initialState = {
+  trades: {},
+}
+
+export const blotterServiceReducer = (state: State = initialState, action): State => {
   switch (action.type) {
-    case ACTION_TYPES.BLOTTER_SERVICE:
-      const newState = state
-      const payloadTrades = action.payload.trades
-
-      // removing the existing key for a single trade
-      if (payloadTrades.length === 1) {
-        const tradeKey = payloadTrades[0]['tradeId']
-        if (state.trades[tradeKey]) {
-          newState.trades = _.omit(state.trades, tradeKey)
-        }
-      }
-      const trades = _.mapKeys(_.values(payloadTrades), 'tradeId')
+    case ACTION_TYPES.BLOTTER_SERVICE_NEW_TRADES:
+      const newTradesById = keyBy(action.payload.trades, `tradeId`)
       return {
```

Comment:
this logic might still be required

Reply:
they had that comment because they are doing different things if one trade is added vs multiple trades are added. The updated code can handle 1 or multiple trades with the same logic. 
Not sure why when they have only one trade, they remove it from the state and then add it again, instead of just replacing it directly.